src/bitcoin-cli.cpp

```diff
@@ -291,6 +292,180 @@ class GetinfoRequestHandler: public BaseRequestHandler
     }
 };
 
+/** Process netinfo requests */
+class NetinfoRequestHandler : public BaseRequestHandler
+{
+private:
+    enum struct m_conn_type {
+        ipv4,
+        ipv6,
+        onion,
+    };
+
+    struct m_peer {
+        int id;
+        int mapped_as;
+        int version;
+        int64_t conn_time;
+        int64_t last_recv;
+        int64_t last_send;
+        double min_ping;
+        double ping;
+        std::string addr;
+        std::string sub_version;
+        m_conn_type conn_type;
+        bool is_block_relay;
+        bool is_outbound;
+        bool operator<(const m_peer& rhs) const { return std::tie(is_outbound, min_ping) < std::tie(rhs.is_outbound, rhs.min_ping); }
+    };
+
+    bool m_verbose{false}; //!< Whether user requested verbose -netinfo report
+
+    /** Whether a peer is an IPv6 connection.
+     * @returns true if addr starts with '['. */
+    bool IsAddrIPv6(const std::string& addr) const { return addr.front() == '['; }
+
+    /** Whether a peer is an outbound onion connection.
+     * @returns true if addr contains ".onion". */
+    bool IsOutboundOnion(const std::string& addr) const { return addr.find(ONION_DOMAIN) != std::string::npos; }
+
+    /** Whether a peer is an inbound onion connection.
+     * @returns true if mapped_as == 0, addr starts with "127.0.0.1", and addr_local contains ".onion". */
+    bool IsInboundOnion(int mapped_as, const std::string& addr, const std::string& addr_local) const
+    {
+        return mapped_as == 0 && addr.find(LOCALHOST_IPV4) == 0 && addr_local.find(ONION_DOMAIN) != std::string::npos;
+    }
+
+    std::string ConnTypeEnumToString(m_conn_type t)
+    {
+        switch (t) {
+        case m_conn_type::ipv4: return "ipv4";
+        case m_conn_type::ipv6: return "ipv6";
+        case m_conn_type::onion: return "onion";
+        } // no default case, so the compiler can warn about missing cases
+        assert(false);
+    }
+
+public:
+    const int ID_PEERINFO = 0;
+    const int ID_NETWORKINFO = 1;
+
+    UniValue PrepareRequest(const std::string& method, const std::vector<std::string>& args) override
+    {
+        if (!args.empty()) {
+            const std::string arg{ToLower(args.at(0))};
+            m_verbose = (arg == "true" || arg == "t" || arg == "1");
```

Comment:
I think policy is now to not allow "1" instead of "true" if it can be avoided. Came up here: https://github.com/bitcoin/bitcoin/pull/19544#discussion_r458695235

Reply:
Interesting, thanks. That may possibly apply more to the RPC API than to the CLI, but ok, done.